### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
- handle the case of `field` attribute request
- refactoring hardcoded strings
- enhance cohesiveness of components
- add `pydantic` to `dev` and `streaming` requirements

### CODE REVIEW

File: pymilo/streaming/communicator.py [hunk 1]
```diff
@@ -34,30 +34,37 @@ def download(self, payload):
...
+        return response.json()["payload"]
 
     def upload(self, payload):
         """
         Upload the local ML model to the remote server.
 
         :param payload: upload request payload
         :type payload: dict
-        :return: response of pymilo server
+        :return: boolean indicating successfulness of upload action
```
Comment (sadrasabouri) [inline pymilo/streaming/communicator.py:50]: `True if upload was successful, False otherwise`

File: pymilo/streaming/pymilo_client.py [hunk 2]
```diff
@@ -57,40 +51,40 @@ def toggle_mode(self, mode=Mode.LOCAL):
...
+        serialized_model = self._communicator.download({
+            "client_id": self._client_id,
             "model_id": self._model_id
         })
-        if response.status_code != 200:
-            print("Remote model download failed.")
-        print("Remote model downloaded successfully.")
-        serialized_model = response.json()["payload"]
+        if serialized_model is None:
+            print("PyMiloClient failed to download the remote ML model.")
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_client.py:68]: move printing strings to params.

File: pymilo/streaming/pymilo_client.py [hunk 3]
```diff
@@ -57,40 +51,40 @@ def toggle_mode(self, mode=Mode.LOCAL):
...
+            return
         self._model = Import(file_adr=None, json_dump=serialized_model).to_model()
-        print("Local model updated successfully.")
+        print("PyMiloClient synched the local ML model with the remote one successfully.")
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_client.py:71]: same.

File: pymilo/streaming/pymilo_client.py [hunk 4]
```diff
@@ -57,40 +51,40 @@ def toggle_mode(self, mode=Mode.LOCAL):
...
             "model_id": self._model_id,
             "model": Export(self._model).to_json(),
         })
-        if response.status_code == 200:
-            print("Local model uploaded successfully.")
+        if succeed:
+            print("PyMiloClient uploaded the local model successfully.")
         else:
-            print("Local model upload failed.")
+            print("PyMiloClient failed to upload the local model.")
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_client.py:87]: same

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thanks for your efforts. Minor comments requested.

**Comment by AHReccese** [comment]
I will update `CHANGELOG.md` soon, review the rest.
@sadrasabouri

**Review by sadrasabouri — APPROVED** [review]
LGTM but introducing new methods should not be under refactoring PR.

**Pull request merged**